File: tb/sim/fibonacci.py
BASE = 0x0001_0000
COUNT = 20

mem_dumps = [range(BASE, BASE + 4 * COUNT)]

def final():
    words = []
    a, b = 1, 1

    for _ in range(COUNT):
        words.append(a)
        c = a + b
        a, b = b, c

    assert_mem(BASE, words)

File: tb/sim/fibonacci.c
void reset()
{
	int a = 1, b = 1, *p = (int*)0x00010000;

	for(int i = 0; i < 20; ++i)
	{
		int c = a + b;

		*p++ = a;
		a = b;
		b = c;
	}
}
